src/EnergyPlus/DataSizing.cc

```diff
@@ -447,86 +445,68 @@ void GetCoilDesFlowT(EnergyPlusData &state,
+            AvgZoneTemp = calcSysSizing.CoolZoneAvgTempSeq(TimeStepAtPeak);
         }
-        DesExitTemp = max(FinalSysSizing(SysNum).CoolSupTemp,
-                          AvgZoneTemp - ZoneCoolLoadSum / (state.dataEnvrn->StdRhoAir * CpAir * FinalSysSizing(SysNum).DesCoolVolFlow));
-        DesFlow = FinalSysSizing(SysNum).DesCoolVolFlow;
+        DesExitTemp =
+            max(finalSysSizing.CoolSupTemp, AvgZoneTemp - ZoneCoolLoadSum / (state.dataEnvrn->StdRhoAir * CpAir * finalSysSizing.DesCoolVolFlow));
+        DesFlow = finalSysSizing.DesCoolVolFlow;
         DesExitHumRat = Psychrometrics::PsyWFnTdbRhPb(state, DesExitTemp, 0.9, state.dataEnvrn->StdBaroPress, "GetCoilDesFlowT");
-    } else if (CoolCapCtrl == Bypass) {
-        if (FinalSysSizing(SysNum).CoolingPeakLoadType == SensibleCoolingLoad) {
-            ZoneCoolLoadSum = CalcSysSizing(SysNum).SumZoneCoolLoadSeq(TimeStepAtPeak);
-            AvgZoneTemp = CalcSysSizing(SysNum).CoolZoneAvgTempSeq(TimeStepAtPeak);
-        } else if (FinalSysSizing(SysNum).CoolingPeakLoadType == TotalCoolingLoad) {
-            ZoneCoolLoadSum = CalcSysSizing(SysNum).SumZoneCoolLoadSeq(TimeStepAtPeak);
-            AvgZoneTemp = CalcSysSizing(SysNum).CoolZoneAvgTempSeq(TimeStepAtPeak);
+    } else if (sysSizInput.CoolCapControl == Bypass) {
+        if (finalSysSizing.CoolingPeakLoadType == SensibleCoolingLoad) {
+            ZoneCoolLoadSum = calcSysSizing.SumZoneCoolLoadSeq(TimeStepAtPeak);
+            AvgZoneTemp = calcSysSizing.CoolZoneAvgTempSeq(TimeStepAtPeak);
+        } else if (finalSysSizing.CoolingPeakLoadType == TotalCoolingLoad) {
+            ZoneCoolLoadSum = calcSysSizing.SumZoneCoolLoadSeq(TimeStepAtPeak);
+            AvgZoneTemp = calcSysSizing.CoolZoneAvgTempSeq(TimeStepAtPeak);
         }
-        AvgSupTemp = AvgZoneTemp - ZoneCoolLoadSum / (state.dataEnvrn->StdRhoAir * CpAir * FinalSysSizing(SysNum).DesCoolVolFlow);
-        TotFlow = FinalSysSizing(SysNum).DesCoolVolFlow;
-        MixTemp = CalcSysSizing(SysNum).MixTempAtCoolPeak;
-        DesExitTemp = FinalSysSizing(SysNum).CoolSupTemp;
-        if (MixTemp > DesExitTemp) {
-            DesFlow = TotFlow * max(0.0, min(1.0, ((MixTemp - AvgSupTemp) / (MixTemp - DesExitTemp))));
+        DesExitTemp = finalSysSizing.CoolSupTemp;
+        if (calcSysSizing.MixTempAtCoolPeak > DesExitTemp) {
+            Real64 AvgSupTemp = AvgZoneTemp - ZoneCoolLoadSum / (state.dataEnvrn->StdRhoAir * CpAir * finalSysSizing.DesCoolVolFlow);
+            DesFlow = finalSysSizing.DesCoolVolFlow *
+                      max(0.0, min(1.0, ((calcSysSizing.MixTempAtCoolPeak - AvgSupTemp) / (calcSysSizing.MixTempAtCoolPeak - DesExitTemp))));
         } else {
-            DesFlow = TotFlow;
+            DesFlow = finalSysSizing.DesCoolVolFlow;
         }
         DesExitHumRat = Psychrometrics::PsyWFnTdbRhPb(state, DesExitTemp, 0.9, state.dataEnvrn->StdBaroPress, "GetCoilDesFlowT");
```

Comment:
Overall changes here are of course the issue correction (2 lines, 456 and 458) and then clean up includes getting rid of locals that could be replaced by simple referenced memory locations and using locals inside specific part of IF blocks instead of setting all the time.